metricbeat/internal/metrics/memory/memory_openbsd.go

```diff
@@ -0,0 +1,236 @@
+	nswapdev           uint32
+	swpages            uint32
+	swpginuse          uint32
+	swpgonly           uint32
+	nswget             uint32
+	nanon              uint32
+	nanonneeded        uint32
+	nfreeanon          uint32
+	faults             uint32
+	traps              uint32
+	intrs              uint32
+	swtch              uint32
+	softs              uint32
+	syscalls           uint32
+	pageins            uint32
+	obsolete_swapins   uint32
+	obsolete_swapouts  uint32
+	pgswapin           uint32
+	pgswapout          uint32
+	forks              uint32
+	forks_ppwait       uint32
+	forks_sharevm      uint32
+	pga_zerohit        uint32
+	pga_zeromiss       uint32
+	zeroaborts         uint32
+	fltnoram           uint32
+	fltnoanon          uint32
+	fltpgwait          uint32
+	fltpgrele          uint32
+	fltrelck           uint32
+	fltrelckok         uint32
+	fltanget           uint32
+	fltanretry         uint32
+	fltamcopy          uint32
+	fltnamap           uint32
+	fltnomap           uint32
+	fltlget            uint32
+	fltget             uint32
+	flt_anon           uint32
+	flt_acow           uint32
+	flt_obj            uint32
+	flt_prcopy         uint32
+	flt_przero         uint32
+	pdwoke             uint32
+	pdrevs             uint32
+	pdswout            uint32
+	pdfreed            uint32
+	pdscans            uint32
+	pdanscan           uint32
+	pdobscan           uint32
+	pdreact            uint32
+	pdbusy             uint32
+	pdpageouts         uint32
+	pdpending          uint32
+	pddeact            uint32
+	pdreanon           uint32
+	pdrevnode          uint32
+	pdrevtext          uint32
+	fpswtch            uint32
+	kmapent            uint32
+}
+
+// Bcachestats reports cache stats from sysctl
+type Bcachestats struct {
+	numbufs        uint64
+	numbufpages    uint64
+	numdirtypages  uint64
+	numcleanpages  uint64
+	pendingwrites  uint64
+	pendingreads   uint64
+	numwrites      uint64
+	numreads       uint64
+	cachehits      uint64
+	busymapped     uint64
+	dmapages       uint64
+	highpages      uint64
+	delwribufs     uint64
+	kvaslots       uint64
+	kvaslots_avail uint64
+}
+
+// Swapent reports swap metrics from sysctl
+type Swapent struct {
+	se_dev      C.dev_t
+	se_flags    int32
+	se_nblks    int32
+	se_inuse    int32
+	se_priority int32
+	sw_path     []byte
+}
+
+func get(_ string) (Memory, error) {
+
+	memData := Memory{}
+
+	n := uintptr(0)
+	var uvmexp Uvmexp
+	mib := [2]int32{C.CTL_VM, C.VM_UVMEXP}
+	n = uintptr(0)
+	// First we determine how much memory we'll need to pass later on (via `n`)
+	_, _, errno := syscall.Syscall6(syscall.SYS___SYSCTL, uintptr(unsafe.Pointer(&mib[0])), 2, 0, uintptr(unsafe.Pointer(&n)), 0, 0)
+	if errno != 0 || n == 0 {
+		return memData, errors.Errorf("Error in size VM_UVMEXP sysctl call, errno %d", errno)
+	}
+
+	_, _, errno = syscall.Syscall6(syscall.SYS___SYSCTL, uintptr(unsafe.Pointer(&mib[0])), 2, uintptr(unsafe.Pointer(&uvmexp)), uintptr(unsafe.Pointer(&n)), 0, 0)
```

Comment:
Should n be the maximum between n and sizeof(uvmexp)? Maybe this way it is possible to continue using the current struct even if this is executed on a future kernel that returns more fields?

Same for bcachestats later.